server/services/zigbee2mqtt/exposes/numericType.js

```diff
@@ -0,0 +1,201 @@
+const { DEVICE_FEATURE_CATEGORIES, DEVICE_FEATURE_TYPES, DEVICE_FEATURE_UNITS } = require('../../../utils/constants');
+
+module.exports = {
+  type: 'numeric',
+  feature: {
+    min: 0,
+    max: 10000,
+  },
+  names: {
+    battery: {
+      feature: {
+        category: DEVICE_FEATURE_CATEGORIES.BATTERY,
+        type: DEVICE_FEATURE_TYPES.SENSOR.INTEGER,
+        unit: DEVICE_FEATURE_UNITS.PERCENT,
+        min: 0,
+        max: 100,
+      },
+    },
+    brightness: {
+      types: {
+        light: {
+          category: DEVICE_FEATURE_CATEGORIES.LIGHT,
+          type: DEVICE_FEATURE_TYPES.LIGHT.BRIGHTNESS,
+          min: 0,
+          max: 255,
+        },
+      },
+    },
+    co2: {
+      feature: {
+        category: DEVICE_FEATURE_CATEGORIES.CO2_SENSOR,
+        type: DEVICE_FEATURE_TYPES.SENSOR.DECIMAL,
+        unit: DEVICE_FEATURE_UNITS.PPM,
+      },
+    },
+    color_temp: {
+      feature: {
+        category: DEVICE_FEATURE_CATEGORIES.LIGHT,
+        type: DEVICE_FEATURE_TYPES.LIGHT.TEMPERATURE,
+        min: 150,
+        max: 500,
+      },
+    },
+    current: {
+      feature: {
+        category: DEVICE_FEATURE_CATEGORIES.SWITCH,
+        type: DEVICE_FEATURE_TYPES.SWITCH.CURRENT,
+        unit: DEVICE_FEATURE_UNITS.AMPERE,
+        min: 0,
+        max: 1000,
+      },
+    },
+    current_phase_b: {
+      feature: {
+        category: DEVICE_FEATURE_CATEGORIES.SWITCH,
+        type: DEVICE_FEATURE_TYPES.SWITCH.CURRENT,
+        unit: DEVICE_FEATURE_UNITS.AMPERE,
+        min: 0,
+        max: 1000,
+      },
+    },
+    current_phase_c: {
+      feature: {
+        category: DEVICE_FEATURE_CATEGORIES.SWITCH,
+        type: DEVICE_FEATURE_TYPES.SWITCH.CURRENT,
+        unit: DEVICE_FEATURE_UNITS.AMPERE,
+        min: 0,
+        max: 1000,
+      },
+    },
+    cpu_temperature: {
+      feature: {
+        category: DEVICE_FEATURE_CATEGORIES.DEVICE_TEMPERATURE_SENSOR,
+        type: DEVICE_FEATURE_TYPES.SENSOR.DECIMAL,
+        unit: DEVICE_FEATURE_UNITS.CELSIUS,
+        min: -100,
+        max: 150,
+      },
+    },
+    device_temperature: {
+      feature: {
+        category: DEVICE_FEATURE_CATEGORIES.DEVICE_TEMPERATURE_SENSOR,
+        type: DEVICE_FEATURE_TYPES.SENSOR.DECIMAL,
+        unit: DEVICE_FEATURE_UNITS.CELSIUS,
+        min: -100,
+        max: 150,
+      },
+    },
+    eco2: {
+      feature: {
+        category: DEVICE_FEATURE_CATEGORIES.CO2_SENSOR,
+        type: DEVICE_FEATURE_TYPES.SENSOR.DECIMAL,
+        unit: DEVICE_FEATURE_UNITS.PPM,
+      },
+    },
+    energy: {
+      feature: {
+        category: DEVICE_FEATURE_CATEGORIES.SWITCH,
+        type: DEVICE_FEATURE_TYPES.SWITCH.ENERGY,
+        unit: DEVICE_FEATURE_UNITS.KILOWATT_HOUR,
+      },
+    },
+    formaldehyd: {
```

Comment:
Just a question, what is this suppose to detect?

Because it's categorized in Gladys as a smoke sensor, is it really a smoke sensor?

Reply:
According to https://github.com/GladysAssistant/Gladys/issues/1246 

> formaldehyd (méthanal ou formaldéhyde ou aldéhyde formique in 🇫🇷)

Maybe I should remove it for now?

Reply:
I removed it.

Reply:
Ok it's a very specific feature. Maybe we could create a new feature category ? (can be in another PR, or this one, as you want)